pkg/controllers/state/suite_test.go

```diff
@@ -333,6 +334,27 @@ var _ = Describe("Inflight Nodes", func() {
 		ExpectResources(instanceType.Allocatable(), ExpectStateNodeExists(node).Allocatable())
 		ExpectResources(instanceType.Capacity, ExpectStateNodeExists(node).Capacity())
 	})
+	It("should populate inflight details once", func() {
+		instanceType := cloudProvider.InstanceTypes[0]
+		node := test.Node(test.NodeOptions{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{
+				v1alpha5.ProvisionerNameLabelKey: provisioner.Name,
+				v1.LabelInstanceTypeStable:       instanceType.Name,
+			}},
+			ProviderID: test.RandomProviderID(),
+		})
+		ExpectApplied(ctx, env.Client, node)
+		ExpectReconcileSucceeded(ctx, nodeController, client.ObjectKeyFromObject(node))
+
+		ExpectStateNodeCount("==", 1)
+		ExpectResources(instanceType.Allocatable(), ExpectStateNodeExists(node).Allocatable())
+		ExpectResources(instanceType.Capacity, ExpectStateNodeExists(node).Capacity())
+
+		cloudProvider.InstanceTypes = cloudProvider.InstanceTypes[1:]
+		// Still expect to succeed w/o instance type since inflight details are set once
+		ExpectReconcileSucceeded(ctx, nodeController, client.ObjectKeyFromObject(node))
+		cloudProvider.InstanceTypes = append([]*cloudprovider.InstanceType{instanceType}, cloudProvider.InstanceTypes...)
```

Comment:
You don't need to add this back in since we re-fill the instance types on the BeforeEach call here https://github.com/aws/karpenter-core/pull/495/files#diff-8eac1a1e4069ba52700c1850c05457d7ff0edf9169df468beb9438918202a5ecR88